### PR TITLE
feat: added endpoint to change user password

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #1099 

Added endpoint to change user password

### CODE REVIEW

File: src/main/java/com/artipie/api/ManageUsers.java [hunk 1]
```diff
@@ -109,7 +119,40 @@ public void remove(final String uid) {
...
-        throw new IllegalStateException(String.format("User %s does not exist", uid));
+        throw new IllegalStateException(String.format("User %s does not exist", uname));
+    }
+
+    @Override
+    public void alterPassword(final String uname, final JsonObject info) {
+        if (this.users().map(yaml -> yaml.yamlMapping(uname) != null).orElse(false)) {
+            YamlMappingBuilder all = Yaml.createYamlMappingBuilder();
+            final YamlMapping users = this.users().get();
+            for (final YamlNode node : users.keys()) {
```
Comment (andpopov) [inline src/main/java/com/artipie/api/ManageUsers.java:130]: Lines 127 and 129 invoke the same method: this.users()
I would to store result and reuse it.

File: src/main/java/com/artipie/api/ManageUsers.java [hunk 2]
```diff
@@ -109,7 +119,40 @@ public void remove(final String uid) {
...
+                final String val = node.asScalar().value();
+                if (!uname.equals(val)) {
```
Comment (andpopov) [inline src/main/java/com/artipie/api/ManageUsers.java:132]: Note no change required: It is just example of usage standard java style of coding, it is not requirement to modify code:
Line 131: for variable 'val' is good name is 'userName'
It is good example for usage CamelCase

File: src/main/java/com/artipie/api/ManageUsers.java [hunk 3]
```diff
@@ -109,7 +119,40 @@ public void remove(final String uid) {
...
+                    all = all.add(val, users.yamlMapping(val));
+                }
+            }
+            final YamlMapping user = this.users().get().yamlMapping(uname);
+            YamlMappingBuilder changing = Yaml.createYamlMappingBuilder();
```
Comment (andpopov) [inline src/main/java/com/artipie/api/ManageUsers.java:137]: Line 136 is the same as : 127 and 129

File: src/main/java/com/artipie/api/ManageUsers.java [hunk 4]
```diff
@@ -109,7 +119,40 @@ public void remove(final String uid) {
...
+            for (final YamlNode node : user.keys()) {
+                final String val = node.asScalar().value();
+                if (!ManageUsers.TYPE.equals(val) && !ManageUsers.PASS.equals(val)) {
```
Comment (andpopov) [inline src/main/java/com/artipie/api/ManageUsers.java:140]: Just information, changing is not mandatory.
Line 139: variable name 'val' is not informative, for example it may be 'key' or 'property'

File: src/main/java/com/artipie/api/ManageUsers.java [hunk 5]
```diff
@@ -109,7 +119,40 @@ public void remove(final String uid) {
...
+            all = all.add(uname, changing.build());
+            this.blsto.save(
+                this.key,
+                Yaml.createYamlMappingBuilder()
+                    .add(YamlSettings.NODE_CREDENTIALS, all.build())
+                    .build().toString().getBytes(StandardCharsets.UTF_8)
+            );
+            return;
+        }
+        throw new IllegalStateException(String.format("User %s does not found", uname));
```
Comment (andpopov) [inline src/main/java/com/artipie/api/ManageUsers.java:155]: Cannot find usage of 'old_pass'

File: src/test/java/com/artipie/api/ManageUsersTest.java [hunk 6]
```diff
@@ -162,6 +162,39 @@ void removesUser() {
         );
     }
 
+    @Test
+    void altersPassword() {
+        this.blsto.save(
```
Comment (andpopov) [inline src/test/java/com/artipie/api/ManageUsersTest.java:167]: 'old_pass' not found usage

**Review by andpopov — DISMISSED** [review]

File: src/test/java/com/artipie/api/ManageUsersTest.java (hunk 6 above)
Comment (olenagerasimova) [reply src/test/java/com/artipie/api/ManageUsersTest.java:167]: check `UsersRest.java:128`

File: src/main/java/com/artipie/api/ManageUsers.java (hunk 5 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/api/ManageUsers.java:155]: check `UsersRest.java:128`, it is not a responsibility of `ManageUsers` class to verify current password, `UsersRest` does it before calling `ManageUsers.alterPassword()` to apply the changes

File: src/main/java/com/artipie/api/ManageUsers.java (hunk 2 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/api/ManageUsers.java:132]: `name` is enough in this case, we work with users here, so it's obvious that `name` is the name of a user

**Comment by olenagerasimova** [comment]
@andpopov thanks, corrected and answered

**Review by andpopov — APPROVED** [review]
@olenagerasimova Thanks!

File: src/test/java/com/artipie/api/ManageUsersTest.java [hunk 7]
```diff
@@ -162,6 +162,39 @@ void removesUser() {
         );
     }
 
+    @Test
+    void altersPassword() {
```
Comment (dgarus) [inline src/test/java/com/artipie/api/ManageUsersTest.java:166]: Think we should have a test that checks the case when the old password is invalid.

**Review by dgarus — CHANGES_REQUESTED** [review]
@olenagerasimova thanks!
We have to add tests for negative cases 404 and 401

**Review by dgarus — APPROVED** [review]
@olenagerasimova Thanks!
LGTM
